### DISCUSSION TITLE
HLS is not accessible through credentials.

### DISCUSSION THREAD

**Discussion by Prakash2101** [bluenviron/mediamtx#616] [Questions]
I have added username and password in yaml file and I'm able to access the rtsp over it but not hls. 
When I looked into the logs over there I got that for 3sec it is getting the m3u8 files and then suddenly it stop. Further I tried by removing the username and password from yaml file and surprisingly hls started working. 
So can you please look into this and give me a solution to access hls url with credentials.  

Thanks.

**Comment by aler9** [comment]
Hello, reading a stream with HLS and credentials works flawlessly in Firefox, Chrome and <s>VLC</s>.
Please provide:

* the name and the version of the browser / reader that is not working
* full server logs

**Comment by Prakash2101** [comment]
Hi @aler9,
I'm using vlc player(version: 3.0.14) for HLS. Sharing below the yaml file in txt and logs file for your reference. 

image used: **aler9/rtsp-simple-server:v0.17..3**


[rtsp-simple-server.txt](https://github.com/aler9/rtsp-simple-server/files/7327264/rtsp-simple-server.txt)
[rtsp-simple-server-logs.log](https://github.com/aler9/rtsp-simple-server/files/7327265/rtsp-simple-server-logs.log)

**Reply by aler9** [reply]
Hello, i checked and as you say it's true that VLC is not able to read HLS streams protected with authentication.

VLC inserts username and password (the Authorization header) into the playlist request:

```
GET /stream/stream.m3u8 HTTP/1.1
Host: localhost:8888
Accept: */*
Accept-Language: en_US
Authorization: Basic dGVzdDp0YXN0
Range: bytes=0-
User-Agent: VLC/3.0.9.2 LibVLC/3.0.9.2
```

But it doesn't seem to insert them into any of the fragment requests:

```
GET /stream/1634484612.ts HTTP/1.1
Host: localhost:8888
Cache-Control: no-cache
User-Agent: VLC/3.0.9.2 LibVLC/3.0.9.2
```

In my opinion you have to [report the issue to the VLC team](https://code.videolan.org/videolan/vlc/-/issues/).

**Reply by Prakash2101** [reply]
@aler9,
I also tried with online player as well, sharing below their url:
https://hls-js.netlify.app/demo/
https://players.akamai.com/players/hlsjs

Can you please suggest some alternative solution for my issue.

**Reply by aler9** [reply]
Hello, i checked HLS with the online players you posted, and all i can say is that they don't support HTTP authentication.
HTTP authentication is the standard way to protect contents served with HTTP; credentials can be passed through the URL:

```
http://myuser:mypass@localhost:8888/mystream/index.m3u8
```

Both VLC and the online players strip the credentials and don't send them to the server.

I don't know why there's so few support around, but you have to report the issue to VLC and hls.js.

In the meanwhile you can use the RTSP protocol with VLC, and protect the HLS player and stream with a reverse proxy and a cookie-based authentication system.

[a comment by an automated account is not shown]
